examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,170 @@
+    mask_images[tissue_name] = image.new_img_like(mask_img, eroded_mask_data)
+
+##########################################################################
+# Next, we resample the tissue masks to the functional image
+from nilearn import image, masking
+mean_func_img = image.mean_img(func_filename)
+func_mask_img = masking.compute_epi_mask(mean_func_img)
+affine = mean_func_img.get_affine()
+shape = mean_func_img.get_data().shape
+for tissue_name, mask_img in mask_images.items():
+    mask_images[tissue_name] = image.resample_img(mask_img, target_shape=shape,
+                                                  target_affine=affine,
+                                                  interpolation='nearest')
+
+##########################################################################
+# and intersect them with a brain mask computed from the mean functional image.
+from nilearn import masking
+func_mask_img = masking.compute_epi_mask(mean_func_img, opening=0)
+for tissue_name, mask_img in mask_images.items():
+    mask_images[tissue_name] = masking.intersect_masks(
+        [func_mask_img, mask_img], threshold=1, connected=False)
+
+##########################################################################
+# We check the masks quality by plotting their contours
+# on top of the mean functional image.
+from nilearn import plotting
+cut_coords = (15, 8, 32)  # corpus callosum
+display = plotting.plot_anat(mean_func_img, cut_coords=cut_coords)
+for tissue_name, colors in zip(['gm', 'wm', 'csf'], 'ymg'):
+    display.add_contours(mask_images[tissue_name], colors=colors)
+
+##########################################################################
+# Now we are ready to compute the voxel-wise signals within the
+# WM and CSF masks, and perform a PCA to extract the top 5 components.
+# All is done by the function **nilearn.image.high_variance_confounds**,
+# with the parameter **percentile** set to **100.**, to include all the voxels
+# of the mask.
+import numpy as np
+for tissue_name in ['wm', 'csf']:
+    tissue_confounds = image.high_variance_confounds(
+        func_filename, mask_img=mask_images[tissue_name], n_confounds=5,
+        percentile=100.)
+    confounds = np.hstack((confounds, tissue_confounds))
+
+#######################################################
+# Exploring confounds contribution to GM voxels signals
+# -----------------------------------------------------
+# Typically motion has a dramatic impact on voxels within the contour of the
+# brain.
+
+#######################################################################
+# We compute GM voxels signals
+from nilearn import input_data
+gm_masker = input_data.NiftiMasker(mask_img=mask_images['gm'],
+                                   memory='nilearn_cache',
+                                   standardize=True)
+gm_signals = gm_masker.fit_transform(func_filename)
+
+#######################################################################
+# and predict GM voxels signals from confounds.
```

Comment:
Can you add a bit of details here basically why you are predicting GM voxels from confounds ?